### PR TITLE
LPS-107816

### PR DESCRIPTION
**Pull request by jesseyeh-liferay**
[LPS-107816](https://issues.liferay.com/browse/LPS-107816)

This changes the type of the version column in KaleoDefinition from an int to a string so as to be consistent with the KaleoDefinitionVersion table.

### CODE REVIEW

**Comment by jesseyeh-liferay** [comment]
ci:test:sf

File: modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -376,7 +404,12 @@ public KaleoDefinition updatedKaleoDefinition(
 		kaleoDefinition.setDescription(description);
 		kaleoDefinition.setContent(content);
 
-		int nextVersion = kaleoDefinition.getVersion() + 1;
+		String version = kaleoDefinition.getVersion();
+
+		int currentVersion = GetterUtil.getInteger(
+			version.substring(0, version.indexOf(StringPool.PERIOD)));
```
Comment (jesseyeh-liferay) [inline modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoDefinitionLocalServiceImpl.java:410]: I'm not too familiar on the relationship between KaleoDefinitionLocalServiceImpl and KaleoDefinitionImpl. This code retrieves the version as a string, but the same logic ends up being repeated across both classes. Is there a better way of handling this? Perhaps in a common *Util class?

**Comment by SamZiemer** [comment]
Hi @jesseyeh-liferay, I'll be reviewing this today or tomorrow, thanks!

**Comment by jesseyeh-liferay** [comment]
Resent: https://github.com/SamZiemer/liferay-portal/pull/797

**Pull request closed without merging**